contracts/smart-account/modules/AccountRecoveryModule.sol

```diff
@@ -460,6 +419,49 @@ contract AccountRecoveryModule is
         emit RecoveryRequestSubmitted(msg.sender, recoveryCallData);
     }
 
+    /**
+     * @dev Executes recovery request for a Smart Account (msg.sender)
+     * Should be called by the Smart Account
+     * SA.execute => AccRecovery.executeRecovery
+     * Decrements recoveries left, and if 0 left, no userOps will be validated by this module
+     * It forces user to perform an explicit action:
+     *      - If user wants same guardians to be able to recover the account again,
+     *      they have to call setAllowedRecoveries() => NOT RECOMMENDED
+     *     - If user wants to change guardians, they have to
+     *          -- remove/replace guardians + adjust threshold + setAllowedRecoveries()
+     *          or
+     *          -- clear all guardians + re-init the module => RECOMMENDED
+     * @param to destination address
+     * @param value value to send
+     * @param data callData to execute
+     */
+    function executeRecovery(
+        address to,
+        uint256 value,
+        bytes calldata data
+    ) public {
+        delete _smartAccountRequests[msg.sender];
+        emit RecoveriesLeft(
+            msg.sender,
+            --_smartAccountSettings[msg.sender].recoveriesLeft
+        );
+        (bool success, bytes memory retData) = ISmartAccount(msg.sender)
```

Comment:
can leave second empty if not using retData

Reply:
retData is used in the `revert` next